contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -1476,22 +1516,37 @@ handleForJsonAuto(Query *query)
 				foreach(lc, subq->targetList) {
 					TargetEntry* te = castNode(TargetEntry, lfirst(lc));
 					int oid = te->resorigtbl;
+					String* s = castNode(String, lfirst(list_nth_cell(colnameAlias->colnames, i)));
+					if(te->expr != NULL && nodeTag(te->expr) == T_SubLink) {
+						SubLink *sl = castNode(SubLink, te->expr);
+						if(sl->subselect != NULL && nodeTag(sl->subselect) == T_Query) {
+							if(handleForJsonAuto(castNode(Query, sl->subselect), tableInfoArr, numTables)) {
+								CoerceViaIO *iocoerce = makeNode(CoerceViaIO);
+								iocoerce->arg = (Expr*) sl;
+								iocoerce->resulttype = T_JsonArrayQueryConstructor;
+								iocoerce->resultcollid = 0;
+								iocoerce->coerceformat = COERCE_EXPLICIT_CAST;
+								buildJsonEntry(1, "temp", te);
+								s->sval = te->resname;
```

Comment:
what's the purpose of this line ?
```
s->sval = te->resname
```

Reply:
We do the same thing for each of the JSON object / column names. This is whats used by the tsql_auto_row_to_json as the modified column name